### PR TITLE
Add sync settings for java dependency viewer

### PR DESCRIPTION
**Pull request by yaohaizh**

### CODE REVIEW

**Comment by testforstephen** [comment]
can you share some background about the purpose of this PR?

**Comment by yaohaizh** [comment]
Use javaConfig.json instead for PDE development. This name is temporarily.

File: src/settings.ts [hunk 1]
```diff
@@ -19,10 +20,13 @@ export class Settings {
                     && updatedConfig.syncWithFolderExplorer)) {
                 this._dependencyConfig = updatedConfig;
                 commands.executeCommand(Commands.VIEW_PACKAGE_REFRESH);
+            } else if (updatedConfig.autoRefresh !== this._dependencyConfig.autoRefresh) {
+                SyncHandler.updateFileWatcher(Settings.autoRefresh());
```
Comment (testforstephen) [inline src/settings.ts]: Changed to `SyncHandler.updateFileWatcher(updatedConfig.autoRefresh)`

File: src/settings.ts [hunk 2]
```diff
@@ -19,10 +20,13 @@ export class Settings {
...
             } else {
                 this._dependencyConfig = updatedConfig;
```
Comment (testforstephen) [inline src/settings.ts]: Move this line to the outer, make sure the cached config is synced with the latest.

**Comment by yaohaizh** [comment]
test this please

**Pull request merged**